folly/test/ReplaceableTest.cpp

```diff
@@ -289,3 +289,11 @@ TEST(ReplaceableTest, Conversions) {
   Replaceable<OddA> rOddA{std::move(rOddB)};
   Replaceable<OddB> rOddB2{rOddA};
 }
+
+#if __cpp_deduction_guides >= 201703
+TEST(ReplaceableTest, DeductionGuide) {
+  Basic b{};
+  Replaceable r{b};
+  static_assert(std::is_same_v<Replaceable<Basic>, decltype(r)>);
```

Comment:
There are other checks of compile-time properties, but they are done as runtime expectations with `EXPECT_EQ` etc.

Reply:
Is there a particular reason to use `EXPECT_EQ` over the `static_assert`? Happy to change, just looking to understand the reasoning.

Reply:
Builds by default tend to terminate at the first build failure, but tests by default show all expectation failures.

Reply:
Just switched to `EXPECT_TRUE`. Want me to do the same for `fbvector` tests over in https://github.com/facebook/folly/pull/1090?

Reply:
Please.